### PR TITLE
Fixes #21009: System update technique does not work as intended

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/21009

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

File: techniques/applications/systemUpdate/1.0/systemUpdate.st [hunk 1]
```diff
@@ -46,25 +46,29 @@ bundle agent rudder_system_update {
...
 
 
-   pass2::
-    "flag_set" expression => strcmp("${is_set[1]}", "set");
-
+    pass2::
+      "flag_set" expression => strcmp("${is_set[1]}", "set");
+    pass3::
+      "must_run_system_update" expression => "any",
+                                       if => "schedule_simple_${canonified_campaign_name}_repaired";
```
Comment (Fdall) [inline techniques/applications/systemUpdate/1.0/systemUpdate.st:63]:
```suggestion
      "must_run_system_update" expression =>  "schedule_simple_${canonified_campaign_name}_repaired";
```

File: techniques/applications/systemUpdate/1.0/systemUpdate.st [hunk 2]
```diff
@@ -76,16 +80,18 @@ bundle agent rudder_system_update {
                                                                             "1", # periodicity days
                                                                             "stateless"
                                                                           );
+      "enable_reporting"   usebundle => enable_reporting;
```
Comment (Fdall) [inline techniques/applications/systemUpdate/1.0/systemUpdate.st:83]: Not needed, it is enabled by the metadata.xml

Comment (VinceMacBuche) [reply techniques/applications/systemUpdate/1.0/systemUpdate.st:83]: it is needed because i disabled it jut above to call a schedule_simple

Comment (VinceMacBuche) [reply techniques/applications/systemUpdate/1.0/systemUpdate.st:83]: and still have the logs of the schedule_simple

**Comment by VinceMacBuche** [comment]
Commit modified

File: techniques/applications/systemUpdate/1.0/systemUpdate.st [hunk 3]
```diff
@@ -76,16 +80,18 @@ bundle agent rudder_system_update {
...
 
       # Pre-Update
+    pass1::
       "Setup variable context" usebundle => _method_reporting_context_v4("Setup variable context", "Setup variable context", "Setup variable context");
-      "Unsupported OS"         usebundle => report_if_condition("variable_reporting", "use_yum_provider|user_apt_provider|user_zypper_provider");
+      "Unsupported OS"         usebundle => report_if_condition("variable_reporting", "use_yum_provider|use_apt_provider|use_zypper_provider");
```
[the remark this answers is not in the document]
Comment (VinceMacBuche) [reply techniques/applications/systemUpdate/1.0/systemUpdate.st:88]: We will fix it in next iteration, keep it for now

**Review by Fdall — APPROVED** [review]

**Pull request merged**